rotkehlchen/exchanges/coinbase.py

```diff
@@ -684,6 +663,97 @@ def _process_normal_trade(self, event: dict[str, Any], trades: list[Trade]) -> N
 
         return None
 
+    def _process_normal_trade(self, event: dict[str, Any], timestamp: Timestamp) -> Trade | None:
+        """Turns a normal coinbase transaction into a rotki trade and returns it.
+
+        Sometimes the amounts can be negative which breaks rotki's logic which is why we use abs().
+
+        https://docs.cloud.coinbase.com/sign-in-with-coinbase/docs/api-transactions#transaction-types
+        If the coinbase transaction is not a trade related transaction nothing happens.
+        """
+        trade_type = TradeType.deserialize(event['type'])
+        amount = AssetAmount(abs(deserialize_asset_amount(event['amount']['amount'])))
+        tx_asset = asset_from_coinbase(event['amount']['currency'], time=timestamp)
+        native_amount = abs(deserialize_asset_amount(event['native_amount']['amount']))
+        native_asset = asset_from_coinbase(event['native_amount']['currency'], time=timestamp)
+        # rate is how much you get/give in quotecurrency if you buy/sell 1 unit of basecurrency
+        rate = Price(native_amount / amount)
```

Comment:
It shouldn't happen but I just noticed that this can raise DivisionByZeroError. Can you also catch it? I recall one exchange raising it once 🙈